video/out/opengl/egl_helpers.c

```diff
@@ -127,15 +152,25 @@ static bool create_context(struct ra_ctx *ctx, EGLDisplay display,
         return false;
     }
 
+    EGLint r_size = opts->output_format >> 24,
+           g_size = (opts->output_format >> 16) & 0xff,
+           b_size = (opts->output_format >> 8) & 0xff,
+           a_size = opts->output_format & 0xff;
+
     EGLint attributes[] = {
         EGL_SURFACE_TYPE, EGL_WINDOW_BIT,
-        EGL_RED_SIZE, 8,
-        EGL_GREEN_SIZE, 8,
-        EGL_BLUE_SIZE, 8,
-        EGL_ALPHA_SIZE, ctx->opts.want_alpha ? 8 : 0,
+        EGL_RED_SIZE, MPMAX(r_size, 8),
+        EGL_GREEN_SIZE, MPMAX(g_size, 8),
+        EGL_BLUE_SIZE, MPMAX(b_size, 8),
+        EGL_ALPHA_SIZE, MPMAX(a_size, ctx->opts.want_alpha ? 8 : 0),
         EGL_RENDERABLE_TYPE, rend,
         EGL_NONE
     };
+    if (opts->config_id) {
+        attributes[0] = EGL_CONFIG_ID;
+        attributes[1] = opts->config_id;
+        attributes[2] = EGL_NONE;
```

Comment:
would make sense to include at least `EGL_SURFACE_TYPE` here too, in case the user specifies something incompatible
maybe even `EGL_RENDERABLE_TYPE` too

Reply:
fixed in 8cb35ea54cf5adc407dfd6420abeefbe72b10d8d